lib/BackgroundJobs/IndexerJob.php

```diff
@@ -145,34 +148,27 @@ protected function getMaxIndexingTime(): int {
 	}
 
 	protected function hasEnoughRunningJobs(): bool {
-		// Sleep a bit randomly to avoid a scenario where all jobs are started at the same time and kill themselves directly
-		sleep(rand(1, 30));
-		if (!$this->jobList->hasReservedJob(static::class)) {
-			// short circuit to false if no jobs are running, yet
-			return false;
-		}
-		$count = 0;
-		foreach ($this->jobList->getJobsIterator(static::class, null, 0) as $job) {
-			// Check if job is running
-			$query = $this->db->getQueryBuilder();
-			$query->select('*')
-				->from('jobs')
-				->where($query->expr()->gt('reserved_at', $query->createNamedParameter($this->timeFactory->getTime() - 6 * 3600, IQueryBuilder::PARAM_INT)))
-				->andWhere($query->expr()->eq('id', $query->createNamedParameter($job->getId(), IQueryBuilder::PARAM_INT)))
-				->setMaxResults(1);
+		// Cout reserved jobs of last period
+		$query = $this->db->getQueryBuilder();
+		$query->select($query->createFunction('COUNT(*)'))
+			->from('jobs')
+			->where($query->expr()->gt('reserved_at', $query->createNamedParameter($this->timeFactory->getTime() - $this->getMaxIndexingTime(), IQueryBuilder::PARAM_INT)))
```

Comment:
the thing is that this query is also using the 30mins, not the 5mins. My original idea was to unify these, so that

- the job runs every Xmins
- hasEnoughRunningJobs checks the last Xmins
- and the job runs at max Xmins

ie. in this case
- the job runs every 5mins
- hasEnoughRunningJobs checks the last 5mins
- and the job runs at max 5mins

Reply:
But I guess we can decouple the frequency from the max time.

Reply:
Your approach makes sense, I just got a bit impatient with the job execution. Tried with 10 minutes as both interval and max time, from the logs it seemed like only one job was executed at a time when there were many scheduled.
Where is the runner for the "timed jobs"?

Reply:
the timed jobs are run by the same cron.php (every 5 mins) sequentially after checking (https://github.com/nextcloud/server/blob/7689588589d54bf74f4118fda57617dd3b38d052/lib/public/BackgroundJob/TimedJob.php#L83) so 10 mins interval with 30 mins max runtime would make healthy concurrency.

Reply:
Then lets do it like this :)